lib/brainstem/presenter.rb

```diff
@@ -185,17 +185,42 @@ def calculate_sort_name_and_direction(user_params = {})
     # Run preloading on the given models.
     def preload_associations!(models, context)
       if models.length > 0
-        association_names_to_preload = context[:requested_associations_hash].values.map(&:method_name).compact
-        association_names_to_preload += configuration[:preloads].to_a
-        # todo: better de-duping here when things might be nested hashes?
-        association_names_to_preload.uniq!
-        if association_names_to_preload.any?
-          reflections = context[:reflections]
-          association_names_to_preload.reject! { |association| !reflections.has_key?(association.to_s) }
-          if association_names_to_preload.any?
-            Brainstem::Presenter.ar_preload(models, association_names_to_preload)
+        reflections = context[:reflections]
+
+        preloads = context[:requested_associations_hash].values.map(&:method_name).compact
```

Comment:
Agreed, the `context` is a bag of data that is passed through primarily as an optimization to avoid calling things like `Brainstem::Presenter.reflections(models.first.class),` over and over again.